lib/open_uri.ml

```diff
@@ -20,8 +20,11 @@ open Xapi_stdext_pervasives.Pervasiveext
 let handle_socket f s = try f s with e -> Backtrace.is_important e ; raise e
 
 let open_tcp f host port =
-  let host_entry = Unix.gethostbyname host in
-  let sockaddr = Unix.ADDR_INET (host_entry.Unix.h_addr_list.(0), port) in
+  let host = Scanf.ksscanf host (fun _ _ -> host) "[%s@]" Fun.id in
+  let he = Unix.getaddrinfo host (string_of_int port) [] in
+  if he = [] then raise Not_found;
```

Comment:
```
let sockaddr = match Unix.getaddrinfo host (string_of_int port) [] with
| [] -> rasie ..
| addr::_ -> addr.Unix.ai_addr
```

You are raising `Not_found`  - is that exception already handled. If not, it would be better to raise a different exception that signals an internal error. `Not_found` is notoriously hard to find if it is not handled.

Reply:
`Not_found` is what's raised by `gethostbyname` in this case. So I kept the same behavior.
I'm okay to change this to a more explicit error.

Reply:
The `match` would make it easy to log the problem before raising it. If `Not_found` was raised before, it should be fine. It seems to me of these cases that can only happen when something is quite wrong already.

Reply:
Okay, I'll add a match to log something :).
Yes this should really never happened, I've encountered this issue because I'm using IPv6 addresses and `gethostbyname` doesn't support it. I don't see how `getaddrinfo` could return an empty list here!